qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/RcsCcsCommonYamlTestSuiteIT.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.rest.yaml;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import com.carrotsearch.randomizedtesting.annotations.TimeoutSuite;
+
+import org.apache.http.HttpHost;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.tests.util.TimeUnits;
+import org.elasticsearch.Version;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.IOUtils;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.FeatureFlag;
+import org.elasticsearch.test.cluster.local.LocalClusterConfigProvider;
+import org.elasticsearch.test.cluster.util.resource.Resource;
+import org.elasticsearch.test.rest.ObjectPath;
+import org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT.TestCandidateAwareClient;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.unmodifiableList;
+import static org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT.CCS_APIS;
+import static org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT.rewrite;
+
+/**
+ * This runner executes test suits against two clusters (a "write" (the remote) cluster and a
+ * "search" cluster) connected via CCS.
+ * The test runner maintains an additional client to the one provided by ESClientYamlSuiteTestCase
+ * That client instance (and a corresponding client only used for administration) is running all API calls
+ * defined in CCS_APIS against the "search" cluster, while all other operations like indexing are performed
+ * using the client running against the "write" cluster.
+ *
+ */
+@TimeoutSuite(millis = 15 * TimeUnits.MINUTE) // to account for slow as hell VMs
+public class RcsCcsCommonYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
```

Comment:
This test class does not extend `CcsCommonYamlTestSuiteIT` and instead uses the static methods from `CcsCommonYamlTestSuiteIT`. I think this is better choice here because the one test is for legacy model without security and the other is for new RCS with sedcurity and hence there test cluster setup is quite different. The yaml file transform methods can be shared via static methods. I think this is easier to reason about compared to thinking how one test class might impact the other.